### PR TITLE
Don't try to start paused webhooks when ensuring that webhooks are running properly

### PR DESCRIPTION
**Pull request by lkemmerer**
This isn't quite ready for review as I need to write up a description and maybe add some comments (and do one last run through CI). I'll add reviewers when it's ready.

### CODE REVIEW

File: src/main/java/com/flightstats/hub/webhook/WebhookLeaderState.java [hunk 1]
```diff
@@ -28,13 +31,40 @@ public ActiveWebhooks(
...
+    public static class RunningState {
+        boolean leadershipAcquired;
+        Set<String> runningServers;
+
+        public boolean isRunningOnSingleServer() {
+            return isLeadershipAcquired() && getRunningServers().size() == 1;
+        }
+
+        public boolean isStopped() {
+            return !isLeadershipAcquired() && getRunningServers().isEmpty();
```
Comment (chriskessel) [inline src/main/java/com/flightstats/hub/webhook/WebhookLeaderState.java:63]: Depending on the construction, runningServers could be null. You could add `@Builder.Default` to the instance declaration.
```
@Builder.Default
Set<String> runningServers() = Collections.emptySet();
```

File: src/main/java/com/flightstats/hub/webhook/InternalWebhookClient.java [hunk 2]
```diff
@@ -45,22 +44,45 @@ boolean stop(String name, String server) {
         return put(server + "/internal/webhook/stop/" + name);
     }
 
-    Optional<String> runOnServerWithFewestWebhooks(String name) {
-        return runOnOneServer(name, getOrderedServers());
+    Optional<String> runOnOnlyOneServer(String name, Collection<String> runningServers) {
```
Comment (lkemmerer) [inline src/main/java/com/flightstats/hub/webhook/InternalWebhookClient.java:47]: This new method does the following:
1.Tries to confirm the webhook is running/started on one of the nodes in the list
1. Sends stop requests to the remaining nodes that were passed in. 
1. If none of the nodes in the list responded that the run is successful, it'll try to fallback to one of the remaining nodes in the cluster that was not in the list.

File: src/main/java/com/flightstats/hub/webhook/InternalWebhookClient.java [hunk 3]
```diff
@@ -45,22 +44,45 @@ boolean stop(String name, String server) {
...
+        return runOnServerWithFewestWebhooksExcluding(name, Collections.emptyList());
+    }
+
+    private Optional<String> runOnOneServer(String name, Stream<String> servers) {
+        return servers
                 .filter(server -> put(server + "/internal/webhook/run/" + name))
                 .findFirst();
     }
 
+    private Optional<String> runOnServerWithFewestWebhooksExcluding(String name, Collection<String> servers) {
```
Comment (lkemmerer) [inline src/main/java/com/flightstats/hub/webhook/InternalWebhookClient.java:75]: I feel like this class has gotten a little out of hand. Maybe there's a way to clean it up, or separate it so there's less going on. ¯\_(ツ)_/¯

**Comment by lkemmerer** [comment]
Going to close this so that I can give some of the commits clearer commit messages, which'll require a force push.

**Pull request closed without merging**